client/src/components/ImportData/zip/ZipFileEntryCard.vue

```diff
@@ -0,0 +1,86 @@
+<script setup lang="ts">
+import { faDatabase, faExclamationTriangle } from "@fortawesome/free-solid-svg-icons";
+import { computed } from "vue";
+
+import type { ImportableFile } from "@/composables/zipExplorer";
+import localize from "@/utils/localization";
+import { bytesToString } from "@/utils/utils";
+
+import GCard from "@/components/Common/GCard.vue";
+
+interface Props {
+    file: ImportableFile;
+    selectable?: boolean;
+    selected?: boolean;
+    gridView?: boolean;
+    bytesLimit?: number;
+}
+
+const props = withDefaults(defineProps<Props>(), {
+    selectable: true,
+    selected: undefined,
+    gridView: undefined,
+    bytesLimit: undefined,
+});
+
+const emit = defineEmits<{
+    (e: "select"): void;
+}>();
+
+const isSelectable = computed(() => {
+    return props.selectable && !sizeLimitExceeded.value;
+});
+
+const sizeLimitExceeded = computed(() => {
+    return props.bytesLimit ? props.file.size > props.bytesLimit : false;
+});
+
+const sizeLimitExceededMessage = computed(() => {
+    return localize("File is too large to import due to browser limitations.");
```

Comment:
```suggestion
    return localize("File is too large to explore due to browser limitations.");
```

Reply:
Hmm, this is really "import" or maybe "upload" :thinking:, this tooltip refers to the "uncompressed" size of a file contained in the ZIP that you are selecting to be uploaded to your history. So, at this point, you already "explored" the ZIP and found this potential file inside to import into Galaxy. 
The uncompressed size limit of 100MB (per file inside the archive) is only applied when the archive is local, because we need to temporarily download the compressed bytes and uncompress them to the browser's IndexDB before uploading the file to Galaxy via the fetch API.

Reply:
You can still upload the zip file, but how I read the message is that the file can't be uploaded at all.
```suggestion
    return localize("File is too large to extract. Please extract on your computer or upload the entire zip file.");
```
It still seems surprising that you can't pass a reader around to do this lazily.

Reply:
Maybe there is a way of streaming and uncompressing the file into IndexDB, I would need to do more research on that...

Reply:
Why is indexdb involved ?

Reply:
We support "exploring" both local ZIP files and URLs to ZIP files. For URLs we don't need to extract anything on the local computer; just use the `zip://` file source handler to point to a file inside the remote ZIP that will be directly downloaded by Galaxy. However, for ZIP files on the local machine, we need a temporary storage to extract individual files before uploading them to Galaxy. 
According to this [Web Storage documentation](https://developer.mozilla.org/en-US/docs/Web/API/Storage_API/Storage_quotas_and_eviction_criteria#web_storage), the browser's "local storage" is very limited (5-10 MiB), but we can have a larger limit using IndexDB or other technologies that depend on the browser's implementation. I also explored the File System API, but this requires the user to interactively designate each file destination, etc. So, IndexDB seemed more convenient.

Reply:
I understand that, but why store the content at all? Can't you pass a reference to the archive reader around?

Reply:
Is something like https://chatgpt.com/share/681a0145-ff08-8007-96b7-7c2131ba3afd possible instead of https://github.com/davelopez/ro-crate-zip-explorer/blob/main/src/zip/localZipService.ts#L30 ?

Reply:
Hmmm, I can try to investigate further how to do it, but it is not just reading arbitrary chunks of the archive; we also need to uncompress the data we read before we get the real file. I'll see what I can do using chunks :+1: